onnxruntime/core/session/inference_session.cc

```diff
@@ -487,10 +488,10 @@ common::Status InferenceSession::Load(std::istream& model_istream) {
   }
 
   auto loader = [this, &model_istream](std::shared_ptr<onnxruntime::Model>& model) {
-    ModelProto model_proto;
+    auto model_proto = onnxruntime::make_unique<ONNX_NAMESPACE::ModelProto>();
```

Comment:
Eventually stored on the heap (https://github.com/microsoft/onnxruntime/blob/c6b499bfc9f92a745b1ab75bc75b698b5b2b9af9/onnxruntime/core/graph/model.cc#L79). Might as well store it on the heap here and pass on the ownership.